apps/omg_api/lib/state/transaction/recovered.ex

```diff
@@ -24,31 +24,52 @@ defmodule OMG.API.State.Transaction.Recovered do
   @empty_signature <<0::size(520)>>
   @type signed_tx_hash_t() :: <<_::768>>
 
-  defstruct [:signed_tx, :signed_tx_hash, spender1: nil, spender2: nil]
+  defstruct [:signed_tx, :signed_tx_hash, spenders: nil]
 
   @type t() :: %__MODULE__{
           signed_tx_hash: signed_tx_hash_t(),
-          spender1: Crypto.address_t() | nil,
-          spender2: Crypto.address_t() | nil,
+          spenders: [Crypto.address_t()],
           signed_tx: Transaction.Signed.t()
         }
 
   @spec recover_from(Transaction.Signed.t()) :: {:ok, t()} | any
-  def recover_from(%Transaction.Signed{raw_tx: raw_tx, sig1: sig1, sig2: sig2} = signed_tx) do
+  def recover_from(%Transaction.Signed{raw_tx: raw_tx, sigs: sigs} = signed_tx) do
     hash_no_spenders = Transaction.hash(raw_tx)
 
-    with {:ok, spender1} <- get_spender(hash_no_spenders, sig1),
-         {:ok, spender2} <- get_spender(hash_no_spenders, sig2),
+    with {:ok, spenders} <- get_spenders(hash_no_spenders, [], sigs),
          do:
            {:ok,
             %__MODULE__{
               signed_tx_hash: Transaction.Signed.signed_hash(signed_tx),
-              spender1: spender1,
-              spender2: spender2,
+              spenders: spenders,
               signed_tx: signed_tx
             }}
   end
 
-  defp get_spender(_hash_no_spenders, @empty_signature), do: {:ok, nil}
-  defp get_spender(hash_no_spenders, sig), do: Crypto.recover_address(hash_no_spenders, sig)
+  defp get_spenders(_hash_no_spenders, acc, []), do: {:ok, Enum.reverse(acc)}
+
+  defp get_spenders(hash_no_spenders, acc, [@empty_signature | sigs]) do
```

Comment:
`Enum.filter(sigs, &not_empty_sig/1) |> Enum.map(&get_spenders/1)`

Reply:
We need to reduce to get a proper error message. I think this rec function is ok, but I can change that to reduce if you think that's more readable.

Reply:
ah ok right. I wish there was an `Enum.` function to handle that in a one-liner.

alternative is

```
result = Enum.filter(sigs, &not_empty_sig/1) |> Enum.map(&get_spenders/1)

case Enum.find(result, check if error) do
  nil -> result
  result -> result
end
```
or similar. I have some preference for avoiding recurrence.

Reply:
ok